gg_analyze/log.py

```diff
@@ -0,0 +1,81 @@
+from datetime import datetime
+import argparse
+import math
+import msgpack
+import numpy as np
+import pandas as pd
+import scipy.interpolate as interp
+import scipy.signal as signal
+import seaborn as sns
+import serial
+import sys
+
+parser = argparse.ArgumentParser(description='Log gufu_gud data.')
+parser.add_argument('device', help='Device name')
+parser.add_argument('--port', '-p', help='Serial port to communicate over')
+
+args = parser.parse_args()
+
+if not args.port:
+    print('Port is required!')
+    sys.exit(1)
+
+if not args.device:
+    print('Device name is required!')
+    sys.exit(1)
+
+filename = f'{args.device}_{datetime.now().strftime("%Y-%m-%d-%H-%M-%S")}.csv'
+
+print("Press Ctrl-C to finish data collection!", flush=True)
+
+# Configure Serial port with 100 ms timeout
+ser = serial.Serial(args.port, timeout=0.1)
+
+# Configure msgpack unpacker for data stream decode
+unpacker = msgpack.Unpacker(raw=False)
+data_list = []
+
+try:
+    while True:
+        # Read some bytes in
+        buf = ser.read(256)
+
+        if not buf:
+            # Try again later
+            continue
+
+        # Feed data to deserialization
+        unpacker.feed(buf)
+
+        try:
+            # Process new objects
+            for obj in unpacker:
+                if not isinstance(obj, dict):
+                    # We only want dicts!
+                    print(f'Rejecting {obj}\t{chr(obj)}', flush=True)
```

Comment:
I would replace this with `print(f'Rejecting {obj}', flush=True)` the chr part was for debug